pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -461,15 +489,17 @@ func (h *Health) checkSystemComponents(
 	shootClient kubernetes.Interface,
 	condition gardencorev1beta1.Condition,
 	extensionConditions []healthchecker.ExtensionCondition,
+	managedResources []resourcesv1alpha1.ManagedResource,
 	healthCheckOutdatedThreshold *metav1.Duration,
 ) (
 	*gardencorev1beta1.Condition,
 	error,
 ) {
-	if exitCondition, err := h.checkManagedResources(ctx, condition, func(managedResource resourcesv1alpha1.ManagedResource) bool {
-		return managedResource.Spec.Class == nil
-	}); err != nil {
-		return exitCondition, err
+	if exitCondition := h.checkManagedResources(condition, managedResources, func(managedResource resourcesv1alpha1.ManagedResource) bool {
+		return managedResource.Spec.Class == nil &&
+			sets.New("", string(gardencorev1beta1.ShootSystemComponentsHealthy)).Has(managedResource.Labels[v1beta1constants.LabelCareConditionType])
```

Comment:
```suggestion
		return managedResource.Spec.Class == nil ||
			sets.New("", string(gardencorev1beta1.ShootSystemComponentsHealthy)).Has(managedResource.Labels[v1beta1constants.LabelCareConditionType])
```

As discussed, let's change this to **OR** to allow MRs with `class=seed` to contribute to `ShootSystemComponentsHealthy`.